akka-actor-typed-tests/src/test/scala/akka/actor/typed/scaladsl/OnSignalSpec.scala

```diff
@@ -29,5 +29,36 @@ final class OnSignalSpec extends ScalaTestWithActorTestKit with AnyWordSpecLike
       spawn[Nothing](behavior)
       probe.expectMessage(Done)
     }
+
+    def stopper(probe: TestProbe[Done]) =
+      Behaviors
+        .receiveMessage[String] {
+          case "stop" =>
+            Behaviors.stopped
+        }
+        .receiveSignal {
+          case (_, PostStop) =>
+            probe.ref ! Done
+            Behaviors.same
+        }
+
+    "execute post stop for child" in {
+      val probe = createTestProbe[Done]("post-stop-child")
+      val stopperRef = spawn(stopper(probe))
+      stopperRef ! "stop"
+      probe.expectMessage(Done)
+
+    }
+
+    "execute post stop for guardian behavior" in {
+      val probe = createTestProbe[Done]("post-stop-probe")
+      val system = ActorSystem(stopper(probe), "work")
+      try {
+        system ! "stop"
+        probe.expectMessage(Done)
+      } finally {
+        system.terminate()
```

Comment:
use this instead
```
ActorTestKit.shutdown(
      system,
      testKitSettings.DefaultActorSystemShutdownTimeout,
      testKitSettings.ThrowOnShutdownTimeout)
```

because it waits for full termination